idlpy/src/context.c

```diff
@@ -530,7 +534,16 @@ static idl_retcode_t write_module_headers(FILE *fh, idlpy_ctx octx, idlpy_module
         idl_fprintf(fh, fmt_entities, entity_prefix, octx->idl_file);
 
         for(int i = 0; i < idlpy_ssos_size(ctx->this_idl_file->entities); ++i) {
-            idl_fprintf(fh, fmt_entity, i > 0 ? ", " : "", idlpy_ssos_at(ctx->this_idl_file->entities, i));
+
+           //Reserved Python keywords support (Issue 105)
+            const char *name = idlpy_ssos_at(ctx->this_idl_file->entities, i);
```

Comment:
If you change `get_c_name` to return the name unmodified when it is not a keyword you can write:
```c
const char *name = get_c_name(idlpy_ssos_at(ctx->this_idl_file->entities, i));
```
However, I suspect the `ctx->this_idl_file->entities` SSOS should contain already filtered names, so here the call is unnecessary?

Reply:
OK, we can do as you suggest, but the cyclonedds-cxx implementation (regarding also using C++ reserved keywords) implements this function (file `idlpy/src/types.c`):

```
const char *get_c_name(const char *name)
{
  //printf("get_c_name: %s \n", name);
  static const size_t n = sizeof(python_keywords)/sizeof(python_keywords[0]);

  /* search through the Python keyword list */
  for (size_t i=0; i < n; i++) {
    if (strcmp(python_keywords[i] + 1, name) == 0)
      return python_keywords[i];
  }

  return NULL;
}
```

**In order to be consistent with other implementations (cxx versus Python), we have followed this example, so, what do you think we have to do at the end?**


Regarding to "_However, I suspect the ctx->this_idl_file->entities SSOS should contain already filtered names, so here the call is unnecessary?_"  

Yes indeed,  `ctx->this_idl_file->entities` does contain the filtered names, so here, the call is unnecessary, in one of our tests, we have worked with this IDL file:

```
module pass {
    struct struct_test {
        char var;
    };
};
```

Our version of the plugin Python IDLC compiler, generates a folder con name "_pass" and inside the files:

```
__init__.py

from ._test import struct_test
__all__ = ["struct_test", ]
```
```
_test.py
from enum import auto
from typing import TYPE_CHECKING, Optional
from dataclasses import dataclass

import cyclonedds.idl as idl
import cyclonedds.idl.annotations as annotate
import cyclonedds.idl.types as types

# root module import for resolving types
import _pass


@dataclass
@annotate.final
@annotate.autoid("sequential")
class struct_test(idl.IdlStruct, typename="_pass.struct_test"):
    var: types.char
```

So, our version is working properly, and we have tested it with a publisher and subscriber in order to check that a message with this format can be sent and received.

Reply:
There is indeed no need to follow the C++ code here. We're going for maximum readability always, if it is used differently in one other place that doesn't really matter to me.